Assets/gantt.js

```diff
@@ -1,112 +1,130 @@
-                    </a>
-                    <table>
-                        <tr>
-                            <td>Start</td>
-                            <td>${start_date}</td>
-                        </tr>
-                        <tr>
-                            <td>End</td>
-                            <td>${end_date}</td>
-                        </tr>
-                        <tr>
-                            <td>Progress</td>
-                            <td>${task.progress}%</td>
-                        </tr>
-                        <tr>
-                            <td>Column</td>
-                            <td>${task.column}</td>
-                        </tr>
-                        <tr>
-                            <td>Swimlane</td>
-                            <td>${task.swimlane}</td>
-                        </tr>
-                        <tr>
-                            <td>Category</td>
-                            <td>${task.category}</td>
-                        </tr>
-                    </table>
-                </div>
-                `;
-      },
-      language: jQuery("html").attr("lang"),
-      on_click: function (task) {
-        //console.log(task);
-        window.open(task.url);
-      },
-      on_date_change: function (task, start, end) {
-        console.log(task, start, end);
-
-        //get save url
-        var sUrl = jQuery("#gantt-chart").data("save-url");
-
-        var oValues = {
-          id: task.id,
-          start: start,
-          end: end,
-        };
-
-        $.ajax({
-          cache: false,
-          url: sUrl,
-          contentType: "application/json",
-          type: "POST",
-          processData: false,
-          data: JSON.stringify(oValues),
+    if (KB.exists("#gantt-chart")) {
+        console.log("load gantt chart");
+
+        var tasks = jQuery("#gantt-chart").data("records");
+        console.log(tasks);
+        var gantt = new Gantt("#gantt-chart", initGanttTasks(tasks), {
+            view_modes: ["Day", "Week", "Month"],
+            view_mode: "Week",
+            date_format: "YYYY-MM-DD",
+            custom_popup_html: function (task) {
+                function format_date(oDate) {
+                    if (oDate.getHours() === 0 && oDate.getMinutes() === 0) {
+                        return oDate.toLocaleDateString();
+                    } else {
+                        return oDate.toLocaleString();
+                    }
+                }
+                const start_date = format_date(task._start);
+                const end_date = format_date(task._end);
+                // TODO: add more info and translate text
+                var html = jQuery('#gantt-popup-template').html();
+                var matches = html.match(/\$\{([\w-_\.]*)\}/g);
+                if (matches) {
+                    jQuery.each(matches, function (i, taskValue) {
+                        var _taskValue = eval(taskValue.slice(2, -1));
```

Comment:
eval is evil 🤔 

maybe just `var _taskValue = task[taskValue.slice(2, -1).split(".")[1]];`

Reply:
That's right ! I corrected that in 065b6bc54c8f60f924c5087cde401ad981df169d
Now, without `eval()`, we can't use multidimensional object like `task.xx.xx`, only `task.xx`. I'll check later how can I manage that.